components/match2/wikis/halo/match_summary.lua

```diff
@@ -102,10 +102,14 @@ function CustomMatchSummary.getByMatchId(args)
 		match.opponents[2].type == Opponent.team
 	then
 		local team1, team2 = string.gsub(match.opponents[1].name, ' ', '_'), string.gsub(match.opponents[2].name, ' ', '_')
-		match.links.headtohead = tostring(mw.uri.fullUrl('Special:RunQuery/Head2head')) ..
-		'?pfRunQueryFormName=Match+history&Head_to_head_query%5Bplayer%5D=' .. team1 ..
-		'&Head_to_head_query%5Bopponent%5D=' .. team2 .. '&wpRunQuery=Run+query'
+		match.links.headtohead = tostring(
+			mw.uri.fullUrl('Special:RunQuery/Head2head')) .. '?title=Special%3ARunQuery%2FHead2head&pfRunQuery' ..
+			'FormName=Head2head&Headtohead%5Bteam1%5D=' .. team1 .. '&Headtohead%5Bteam2%5D=' .. team2..
+			'&Headtohead%5Bgames%5D%5Bis_list%5D=1&Headtohead%5Btiers%5D%5Bis_list%5D=1&Headtohead%5Bfromdate%' ..
+			'5D%5Bday%5D=01&Headtohead%5Bfromdate%5D%5Bmonth%5D=01&Headtohead%5Bfromdate%5D%5Byear%5D='..
+			string.sub(match.date,1,4) ..  '&pf_free_text=&wpRunQuery=Run+query'
```

Comment:
It seems like there is no way to retrieve a plain link from that parser function, it will always return a wikilink (which, in this case, has the default display "Run query".
So you can either add `['link text']=[[File:File:Match Info Stats.png|15px|Head-to-head statistics|link=]]` and add the headtohead manually to the footer, or you'll have to do the manual link creation like you did before.

Do note that you'll only need a link like this one:
`https://liquipedia.net/halo/Special:RunQuery/Head2head?Headtohead%5Bfromdate%5D%5Bday%5D=01&Headtohead%5Bfromdate%5D%5Bmonth%5D=01&Headtohead%5Bfromdate%5D%5Byear%5D=2017&Headtohead%5Bgames%5D%5Bis_list%5D=1&Headtohead%5Bteam1%5D=team1&Headtohead%5Bteam2%5D=team2&Headtohead%5Btiers%5D%5Bis_list%5D=1&_run=`

Reply:
Tried to build something clean:
```suggestion
		buildQueryFormLink = function(form, template, arguments)
			return tostring(mw.uri.fullUrl('Special:RunQuery/' .. form,
				mw.uri.buildQueryString(Table.map(arguments, function(key, value) return template .. key, value end))
			))
		end

		local headtoheadArgs = {
			['[team1]'] = team1,
			['[team2]'] = team2,
			['[games][is_list]'] = 1,
			['[tiers][is_list]'] = 1,
			['[fromdate][day]'] = '01',
			['[fromdate][month]'] = '01',
			['[fromdate][year]'] = string.sub(match.date,1,4),
			['_run'] = false
		}

		match.links.headtohead = buildQueryFormLink('Head2head', 'Headtohead', headtoheadArgs)
```

Reply:
This works. Thank you for testing this too!